front_end/panels/react_devtools/ReactDevToolsView.ts

```diff
@@ -31,6 +35,45 @@ type ContextCreatedEvent = Common.EventTarget.EventTargetEvent<SDK.RuntimeModel.
 // See https://github.com/facebook/react-native/blob/40b54ee671e593d125630391119b880aebc8393d/packages/react-native/ReactCommon/jsinspector-modern/InstanceTarget.cpp#L61
 const MAIN_EXECUTION_CONTEXT_NAME = 'main';
 
+// Based on ExtensionServer.onOpenResource
+async function openResource(
+  url: Platform.DevToolsPath.UrlString,
+  lineNumber: number, // 0-based
+  columnNumber: number, // 0-based
+): Promise<void> {
+  const uiSourceCode = Workspace.Workspace.WorkspaceImpl.instance().uiSourceCodeForURL(url);
+  if (uiSourceCode) {
+    // Unlike the Extension API's version of openResource, we want to normalize the location
+    // so that source maps (if any) are applied.
+    const normalizedUiLocation = await Bindings.DebuggerWorkspaceBinding.DebuggerWorkspaceBinding.instance().normalizeUILocation(uiSourceCode.uiLocation(lineNumber, columnNumber));
+    void Common.Revealer.reveal(normalizedUiLocation);
+    return;
```

Comment:
Will we still need this, if RDT performed symbolication on its side?

Reply:
I doubt that RDT should perform symbolication. That's slower (separate fetch) and potentially less reliable (different fetching mechanism to what CDT uses internally, different source map parsing / semantics / capabilities). When RDT is hosted in an environment that is source map-aware, it should inherit that awareness rather than reimplementing it from scratch.

Reply:
> I doubt that RDT should perform symbolication. That's slower (separate fetch) and potentially less reliable (different fetching mechanism to what CDT uses internally, different source map parsing / semantics / capabilities). When RDT is hosted in an environment that is source map-aware, it should inherit that awareness rather than reimplementing it from scratch.

Then we should at least teach RDT to ask to perform symbolication from source map-aware environment.